compiler-rt/lib/builtins/CMakeLists.txt

```diff
@@ -960,9 +960,9 @@ else ()
         list(APPEND BUILTIN_CFLAGS_${arch} -fomit-frame-pointer -DCOMPILER_RT_ARMHF_TARGET)
       endif()
 
-      # For RISCV32, we must force enable int128 for compiling long
+      # For RISCV32 and 32-bit SPARC, we must force enable int128 for compiling long
       # double routines.
-      if(COMPILER_RT_ENABLE_SOFTWARE_INT128 OR "${arch}" STREQUAL "riscv32")
+      if(COMPILER_RT_ENABLE_SOFTWARE_INT128 OR "${arch}" MATCHES "riscv32|sparc$" AND NOT CMAKE_COMPILER_IS_GNUCC)
```

Comment:
Why is it necessary and why `NOT CMAKE_COMPILER_IS_GNUCC`? Won't it change the behavior for RISCV?

Reply:
Because GCC on SPARC doesn't have `-fforce-enable-int128`. 
I suppose I should change it to `OR "${arch}" STREQUAL "riscv32" OR ("${arch}" STREQUAL "sparc" AND NOT CMAKE_COMPILER_IS_GNUCC)` to not change te RISCV part then?
Not really familiar with how CMake works...